File: frontend/src/types/sourcecontrol.ts
export interface SourceControlAccount {
  id: string
  member_id?: string
  organization_id?: string
  provider_name: string
  provider_id: string
  username: string
  last_synced_at?: string
}

export interface PullRequest {
  id: string
  title: string
  description: string
  url: string
  status: string
  created_at: string
  merged_at?: string
  comments: number
  review_comments: number
  additions: number
  deletions: number
  changed_files: number
}

export interface GetMemberPullRequestsParams {
  startDate?: string
  endDate?: string
}

export interface GetMemberPullRequestReviewsParams {
  startDate?: string
  endDate?: string
}

export interface MemberActivity {
  id: string
  type: 'pull_request' | 'pr_review' | 'pr_comment'
  title: string
  description?: string
  url?: string
  repository?: string
  created_at: string
  metadata?: Record<string, any>
  author_username?: string
  pr_title?: string          // For comments/reviews: the PR title
  pr_author_username?: string // For comments/reviews: the PR author
  pr_metrics?: Record<string, any> // PR performance metrics
}

export interface GetMemberActivityParams {
  startDate?: string // YYYY-MM-DD format
  endDate?: string   // YYYY-MM-DD format
} 

File: backend/http/types/sourcecontrol/sourcecontrol.go
package sourcecontrol

import (
	"time"

	sourcecontroltypes "ems.dev/backend/services/sourcecontrol/types"
)

type ListOrganizationPullRequestsQuery struct {
	UserIDs        []string `form:"userIds" binding:"omitempty"`
	RepositoryName string   `form:"repositoryName" binding:"omitempty"`
	Prefix         string   `form:"prefix" binding:"omitempty"`
	StartDate      string   `form:"startDate" binding:"omitempty,datetime=2006-01-02"`
	EndDate        string   `form:"endDate" binding:"omitempty,datetime=2006-01-02"`
	Status         string   `form:"status" binding:"omitempty,oneof=open closed merged"`
}

type ListOrganizationPullRequestsResponse struct {
	PullRequests []PullRequest `json:"pull_requests"`
}

type PullRequestAuthor struct {
	ID       string `json:"id,omitempty"`
	Username string `json:"username,omitempty"`
	Email    string `json:"email,omitempty"`
}

type PullRequest struct {
	ID             string              `json:"id"`
	Title          string              `json:"title"`
	Description    string              `json:"description"`
	URL            string              `json:"url"`
	Status         string              `json:"status"`
	RepositoryName string              `json:"repository_name"`
	CreatedAt      time.Time           `json:"created_at"`
	MergedAt       *time.Time          `json:"merged_at,omitempty"`
	Comments       int                 `json:"comments"`
	ReviewComments int                 `json:"review_comments"`
	Additions      int                 `json:"additions"`
	Deletions      int                 `json:"deletions"`
	ChangedFiles   int                 `json:"changed_files"`
	Author         *PullRequestAuthor  `json:"author,omitempty"`
}

type ListOrganizationPullRequestsMetricsQuery struct {
	UserIDs        []string `form:"userIds" binding:"omitempty"`
	RepositoryName string   `form:"repositoryName" binding:"omitempty"`
	StartDate      string   `form:"startDate" binding:"omitempty,datetime=2006-01-02"`
	EndDate        string   `form:"endDate" binding:"omitempty,datetime=2006-01-02"`
}

type ListOrganizationSourceControlAccountsResponse struct {
	SourceControlAccounts []sourcecontroltypes.SourceControlAccount `json:"source_control_accounts"`
}

type GetMemberPullRequestsQuery struct {
	StartDate string `form:"startDate" binding:"omitempty,datetime=2006-01-02"`
	EndDate   string `form:"endDate" binding:"omitempty,datetime=2006-01-02"`
}

type GetMemberPullRequestsResponse struct {
	PullRequests []PullRequest `json:"pull_requests"`
}

type GetMemberPullRequestReviewsQuery struct {
	StartDate string `form:"startDate" binding:"omitempty,datetime=2006-01-02"`
	EndDate   string `form:"endDate" binding:"omitempty,datetime=2006-01-02"`
}

type GetMemberPullRequestReviewsResponse struct {
	Reviews []MemberActivity `json:"reviews"`
}

type GetOrganizationMetricsQuery struct {
	StartDate string   `form:"startDate" binding:"omitempty,datetime=2006-01-02"`
	EndDate   string   `form:"endDate" binding:"omitempty,datetime=2006-01-02"`
	Interval  string   `form:"interval" binding:"omitempty,oneof=daily weekly monthly"`
	TeamIDs   []string `form:"teamIds" binding:"omitempty"`
}
